File: python/good_day_name.py
#!/usr/bin/env python3
"""
Format a "Good day" greeting using last name then first name.

Args:
    first_name: The first name of the person.
    last_name: The last name of the person.

Returns:
    A string of the greeting.
"""                                                     
def good_day(first_name: str, last_name: str) -> str:
    if not isinstance(first_name, str) or not first_name.strip():
        raise ValueError("`first_name` must be a non-empty string.")
    if not isinstance(last_name, str) or not last_name.strip():
        raise ValueError("`last_name` must be a non-empty string.")

    first = first_name.strip()
    last = last_name.strip()
    return f"Good day {last} {first}"


if __name__ == "__main__":
    import sys

    try:
        if len(sys.argv) >= 3:
            first_arg = sys.argv[1]
            last_arg = sys.argv[2]
            print(good_day(first_arg, last_arg))
        else:
            first_input = input("Enter first name: ")
            last_input = input("Enter last name: ")
            print(good_day(first_input, last_input))
    except ValueError as err:
        print(err)
        raise SystemExit(1)

File: python/good_day_name.js
const readline = require("readline");
const { goodDay } = require("../javascript/good_day_name.js");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

function askQuestion(query) {
  return new Promise((resolve) => {
    rl.question(query, (answer) => resolve(answer));
  });
}

async function main() {
  const [, , firstArg, lastArg] = process.argv;

  try {
    if (firstArg && lastArg) {
      console.log(goodDay(firstArg, lastArg));
      return;
    }

    const firstInput = await askQuestion("Enter first name: ");
    const lastInput = await askQuestion("Enter last name: ");
    console.log(goodDay(firstInput, lastInput));
  } catch (err) {
    const message = err instanceof Error ? err.message : String(err);
    console.error(message);
    process.exitCode = 1;
  } finally {
    rl.close();
  }
}

main();

